src/pykeen/trackers/file.py

```diff
@@ -0,0 +1,69 @@
+"""Tracking results in local files."""
+
+import logging
+import pathlib
+from typing import Any, Mapping, Optional, TextIO, Union
+
+from .base import ResultTracker
+
+__all__ = [
+    "CSVResultTracker",
+]
+
+logger = logging.getLogger(__name__)
+
+
+class CSVResultTracker(ResultTracker):
+    """
+    Tracking results to a CSV file.
+
+    Also allows monitoring experiments, e.g. by
+
+    .. code ::
+        tail -f results.txt | grep "hits_at_10"
+    """
+
+    #: The file where the results are written to.
+    file: TextIO
+
+    def __init__(self, path: Union[None, str, pathlib.Path] = None):
+        if path is None:
+            path = "results.txt"
+        self.path = pathlib.Path(path)
+        self.path.parent.mkdir(exist_ok=True, parents=True)
+        self.file = self.path.open(mode="w")
```

Comment:
maybe append mode would be better in case the file exists already?

Reply:
I changed it to use a (relatively) unique default name if not path is provided, but still to replace the file.